packages/firestore/test/unit/specs/existence_filter_spec.test.ts

```diff
@@ -286,4 +240,136 @@ describeSpec('Existence Filters:', [], () => {
       );
     }
   );
+
+  /**
+   * ExistenceFilter with BloomFilter
+   * This bloomFilter bitmap size and hashCount is small enough to make false positive rate
+   * as high as 0.5.
+   * {
+   *   bits: {
+   *   bitmap: 'AQ=='
```

Comment:
`documents that mightContain() should return true` is quite easy as it is basically creating a bloomFilter like backend, `documents that mightContain() should return false` might be hard with small bit size(high false positive rate).

Reply:
Maybe we should implement the setBit method in BloomFilter class for testing purposes. Like has an overloading constructor method that takes in hashCount and bitmap size, and create the bitmap based on the strings applied.

Reply:
Here is the method that you could write to generate the bloom filter:

```
function generateBloomFilter(config: {
  contains: Array<string>,
  notContains: Array<string>,
  numBits?: number,
  hashCount?: number,
  padding?: number
}): BloomFilterProto;
```

Then add a `create()` factory method to `BloomFilter` that takes the same arguments as the constructor except it takes the list of strings and the bitmap size (in bytes) instead of the bitmap.

Reply:
For simplicity, I decided to go with "return a bloom filter proto if applicable, or return null", instead of incrementing numBits/hashCount silently till a bloom filter is generated. For spec tests, it should be pretty easy to manually adjust the config to get a desired bloom filter, no big necessity to automize it.

Reply:
Based on the implementation of generateBloomFilter above, removed the "padding" as we are not automatically incrementing the numBits to generate bloom filter. 
To manipulate the padding, we can adjust the numBits(padding = numBits%8).

Reply:
Excellent point. And it makes the implementation much easier. I like it! Although, consider throwing an exception rather than returning null. The exception will provide much better context to the error.